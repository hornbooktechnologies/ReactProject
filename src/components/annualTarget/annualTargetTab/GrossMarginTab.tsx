import React,{ useState } from "react";
import { useTranslation } from "react-i18next";
import AnnualGraphLayout from "../AnnualGraphLayout";
import { getCutoffFractionTime } from '../../../utils/utils';
import { AnnualTargetConstant ,ChartGradientType,ChartColors, AppConstant ,ChartType} from "../../../utils/AppConstants";
import { AnnualGraphLayoutModel } from "../../../models/annualTarget/AnnualGraphLayoutModel";
import GetDonutChartSeriesAndValueForAnnualTarget from "../../../utils/AnnualTargetutils";
import { DonutChartCustomData } from "../../../models/DonutChartDataModel";
import { AnnualTargetDataModel } from "../../../models/annualTarget/AnnualTargetDataModel";

const GrossMarginTab = (props: AnnualTargetDataModel) => {
  const { i18n } = useTranslation();
  const nodataDate = getCutoffFractionTime();
  const [time, setNoDataDate] = useState(nodataDate);
  const [freqValue, setFreqValue] = useState(AnnualTargetConstant.ANNUAL_DATA);
  let ElectricEnergySold = props.jsonData?.GenerationOutput;
  let Spread = props.jsonData?.Spread;
  let Availability = props.jsonData?.Availability;
  let ThermalEfficiency = props.jsonData?.ThermalEfficiency;
  let HeatRate = props.jsonData?.HeatRate;

  const noDataHandler = (val: number, selectValue: string) => {
    setNoDataDate(val);
    setFreqValue(selectValue);
  };

  let ElectricEnergySoldCustomData: DonutChartCustomData = {
    type: ChartType.LARGE,
    displayPage: AnnualTargetConstant.ANNUAL_TARGET,
    color: ChartColors._FFB200,
    header: i18n.t("LABELS.ELECT_ENG_SOLD"),
    chartGradientType: ChartGradientType.BLUE,
    gradientColor1: ChartColors._54AFDE,
    gradientColor2: ChartColors._007CBB
  };
  let ElectricEnergySoldData = GetDonutChartSeriesAndValueForAnnualTarget(
    i18n,
    ElectricEnergySoldCustomData,
    ElectricEnergySold
  );

  let SpreadCustomData: DonutChartCustomData = {
    type: ChartType.LARGE,
    displayPage: AnnualTargetConstant.ANNUAL_TARGET,
    color: ChartColors._FFB200,
    header: i18n.t("ANNUAL_TARGET.SPREAD"),
    chartGradientType: ChartGradientType.GREEN,
    gradientColor1: ChartColors._9FCD72,
    gradientColor2: ChartColors._66A428,
  };
  let SpreadData = GetDonutChartSeriesAndValueForAnnualTarget(
    i18n,
    SpreadCustomData,
    Spread
  );
  let AvailabilityCustomData: DonutChartCustomData = {
    type: ChartType.LARGE,
    displayPage: AnnualTargetConstant.ANNUAL_TARGET,
    color: ChartColors._FFB200,
    header: i18n.t("ANNUAL_TARGET.AVAILABLITY"),
    chartGradientType: ChartGradientType.STRONG_RED,
    gradientColor1: ChartColors._E88000,
    gradientColor2: ChartColors._D82531,
  };
  let AvailabilityData = GetDonutChartSeriesAndValueForAnnualTarget(
    i18n,
    AvailabilityCustomData,
    Availability
  );
  let ThermalEfficiencyCustomData: DonutChartCustomData = {
    type: ChartType.LARGE,
    displayPage: AnnualTargetConstant.ANNUAL_TARGET,
    color: ChartColors._FFB200,
    header: i18n.t("LABELS.THERMAL_EFFICIENCY"),
    chartGradientType: ChartGradientType.GREEN,
    gradientColor1: ChartColors._9FCD72,
    gradientColor2: ChartColors._66A428,
  };
  let ThermalEfficiencyData = GetDonutChartSeriesAndValueForAnnualTarget(
    i18n,
    ThermalEfficiencyCustomData,
    ThermalEfficiency
  );
  let HeatRateCustomData: DonutChartCustomData = {
    type: ChartType.LARGE,
    displayPage: AnnualTargetConstant.ANNUAL_TARGET,
    color: ChartColors._FFB200,
    header: i18n.t("ANNUAL_TARGET.HEAT_RATE"),
    chartGradientType: ChartGradientType.GREEN,
    gradientColor1: ChartColors._9FCD72,
    gradientColor2: ChartColors._66A428,
  };
  let HeatRateData = GetDonutChartSeriesAndValueForAnnualTarget(
    i18n,
    HeatRateCustomData,
    HeatRate
  );

  const graphData: AnnualGraphLayoutModel = {
    GraphLayout: [
      {
        lineData: ElectricEnergySold,
        donutchartData: ElectricEnergySoldData,
        name: AnnualTargetConstant.ELECTRIC_ENERGY_SOLD,
        cumulativeBoolean: false,
      },
      {
        lineData: Spread,
        donutchartData: SpreadData,
        name: AnnualTargetConstant.SPREAD,
        cumulativeBoolean: true,
      },
      {
        lineData: Availability,
        donutchartData: AvailabilityData,
        name: AnnualTargetConstant.AVAILABILITY,
        cumulativeBoolean: true,
      },
      {
        lineData: ThermalEfficiency,
        donutchartData: ThermalEfficiencyData,
        name: AnnualTargetConstant.THERMAL_EFFICIENCY,
        cumulativeBoolean: true,
      },
      {
        lineData: HeatRate,
        donutchartData: HeatRateData,
        name: AnnualTargetConstant.HEAT_RATE,
        cumulativeBoolean: true,
      },
    ],
    noDataHandler: noDataHandler,
    time: time,
    freqValue: freqValue,
  };

  return (
    <>
      <li
        className={`${
          props.AnnualTargetSelectedTab === AppConstant.GROSSMARGIN ? "uk-active" : ""
        }`}
      >
        <AnnualGraphLayout {...graphData} />
      </li>
    </>
  );
};
export default GrossMarginTab;
