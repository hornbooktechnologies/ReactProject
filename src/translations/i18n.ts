import { initReactI18next } from 'react-i18next'
import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import { TRANSLATIONS_EN } from './en/Translations'
import { TRANSLATIONS_JA } from './ja/Translations'

i18n
  .use(LanguageDetector)
  .use(initReactI18next)
  .init({
    lng: 'en',
    resources: {
      en: {
        translation: TRANSLATIONS_EN,
      },
      ja: {
        translation: TRANSLATIONS_JA,
      },
    },
  })

export default i18n
