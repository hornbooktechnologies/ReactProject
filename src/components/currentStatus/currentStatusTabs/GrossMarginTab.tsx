import React from 'react'
import { ChartColors, ChartGradientType, ChartType, AppConstant } from '../../../utils/AppConstants'
import { DonutChartCustomData } from '../../../models/DonutChartDataModel';
import i18n from '../../../translations/i18n';
import { GetDonutChartSeriesAndValueForCurrentStatus } from '../../../utils/utils';
import GrossMarginTabDonutChart from '../currentStatusCharts/GrossMarginTabDonutChart';
import BarRechart from '../../common/AnnualBar';

const GrossMarginTab = ({ annualBarData, GrossMargin } : any): JSX.Element => {
  const pieChartRenderSize = 150;
  let grossMarginCustomData: DonutChartCustomData = {
      header: i18n.t("CURRENT_STATUS.GROSS_MARGIN"),
      type: ChartType.LARGE,
      color: ChartColors._66A428,
      chartGradientType: ChartGradientType.GREEN,
      gradientColor1: ChartColors._9FCD72,
      gradientColor2: ChartColors._66A428,
  }

  let grossMarginData = GetDonutChartSeriesAndValueForCurrentStatus(i18n, grossMarginCustomData, GrossMargin);
  const colorGradient ={
    gradientColor1: ChartColors._9FCD72,
    gradientColor2: ChartColors._66A428,
  }

  return (
      <>
        <li className="uk-active">
          <div className="uk-grid-match pb-20 uk-grid" uk-grid="true">
                <div className="uk-margin-bottom@s uk-width-1-2@m uk-first-column">
                  <div className="uk-card uk-height-1-1 uk-text-center">
                    <BarRechart data={annualBarData} height={192} tickCount={4} colorGradient={colorGradient} type={AppConstant.GROSSMARGIN} barSize={20}></BarRechart>
                  </div>
                </div>
                <div className="uk-margin-bottom@s uk-width-1-2@m">
                  <div className="uk-card uk-card-body uk-flex uk-flex-column">
                    <h6 className="font-16">{grossMarginData.header}</h6>
                    <GrossMarginTabDonutChart 
                      data={grossMarginData} 
                      actualData={GrossMargin}
                      width={pieChartRenderSize}
                      height={pieChartRenderSize}
                    />
                  </div>
                </div>
              </div>
        </li>
      </>
  );
}

export default GrossMarginTab