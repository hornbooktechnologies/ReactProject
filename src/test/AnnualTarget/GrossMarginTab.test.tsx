import React from "react";
import renderer from "react-test-renderer";
import GrossMarginTab from "../../components/annualTarget/annualTargetTab/GrossMarginTab";
import i18n from "../../i18nForTests";
import { I18nextProvider } from "react-i18next";

describe("Component: GrossMarginTab", () => {
  it("GrossMarginTab renders correctly", () => {
    const tree = renderer
      .create(
        <I18nextProvider i18n={i18n}>
          <GrossMarginTab />
        </I18nextProvider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
