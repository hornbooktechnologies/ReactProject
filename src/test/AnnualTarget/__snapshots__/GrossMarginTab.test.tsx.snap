// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: GrossMarginTab GrossMarginTab renders correctly 1`] = `
<li
  className=""
>
  <div
    className="mt-24"
  >
    <h6
      className="m-0"
    >
      LABELS.ELECT_ENG_SOLD
    </h6>
    <div
      className="uk-card mt-24"
      uk-card="true"
    >
      <div
        className="uk-grid-small uk-grid ml-0"
        uk-grid="true"
      >
        <div
          className="uk-margin-bottom@s uk-width-3-4@m uk-padding-remove-left"
        >
          <div
            className="uk-flex-middle"
          />
        </div>
        <div
          className="uk-margin-bottom@s uk-width-1-4@m uk-padding-small mt-50"
        >
          <div
            className="uk-flex"
          >
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.PLAN
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.FORECAST
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
          </div>
          <div
            className="uk-flex uk-flex-center undefined"
          >
            <div
              className="mt-30"
            >
              <div
                className="uk-flex uk-flex-center"
              >
                <div
                  className="chart-continer"
                  style={
                    Object {
                      "height": 170,
                      "width": 170,
                    }
                  }
                >
                  <div
                    className="dount-chart-container"
                  >
                    <div />
                  </div>
                  <div
                    className=" donutmiddleData"
                  >
                    <div
                      className="uk-text-center thumb-font"
                    >
                      <div
                        className="jera-dislike-font"
                      >
                        <i
                          className="jera color-red-60 jera-dislike"
                        />
                      </div>
                      <div
                        className="font-prvalue color-red-60"
                      >
                        <span>
                          0
                        </span>
                        <span
                          className="font-12 vertical-top"
                        >
                          %
                        </span>
                      </div>
                      <div>
                        <span
                          className="font-12 totalRatio"
                        >
                          CHARTS.DONUT.TOOLTIP_HEAD
                        </span>
                        <span
                          className="uk-tooltip-cutom tooltip-text"
                        >
                          <i
                            className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                          />
                          <div
                            className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                          >
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              <strong
                                className="color-grey-blue-95"
                              >
                                CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                              </strong>
                              Â  
                              CHARTS.DONUT.TOOLTIP_SUB_TEXT
                            </span>
                            <span
                              className="color-red-60 mt-15 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION
                            </span>
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION_LHS
                               = Y (
                              LABELS.FORECAST
                              ) / X (
                              LABELS.PLAN
                              )
                            </span>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-24"
  >
    <h6
      className="m-0"
    >
      ANNUAL_TARGET.SPREAD
    </h6>
    <div
      className="uk-card mt-24"
      uk-card="true"
    >
      <div
        className="uk-grid-small uk-grid ml-0"
        uk-grid="true"
      >
        <div
          className="uk-margin-bottom@s uk-width-3-4@m uk-padding-remove-left"
        >
          <div
            className="uk-flex-middle"
          />
        </div>
        <div
          className="uk-margin-bottom@s uk-width-1-4@m uk-padding-small mt-50"
        >
          <div
            className="uk-flex"
          >
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.PLAN
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.FORECAST
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
          </div>
          <div
            className="uk-flex uk-flex-center undefined"
          >
            <div
              className="mt-30"
            >
              <div
                className="uk-flex uk-flex-center"
              >
                <div
                  className="chart-continer"
                  style={
                    Object {
                      "height": 170,
                      "width": 170,
                    }
                  }
                >
                  <div
                    className="dount-chart-container"
                  >
                    <div />
                  </div>
                  <div
                    className=" donutmiddleData"
                  >
                    <div
                      className="uk-text-center thumb-font"
                    >
                      <div
                        className="jera-dislike-font"
                      >
                        <i
                          className="jera color-red-60 jera-dislike"
                        />
                      </div>
                      <div
                        className="font-prvalue color-red-60"
                      >
                        <span>
                          0
                        </span>
                        <span
                          className="font-12 vertical-top"
                        >
                          %
                        </span>
                      </div>
                      <div>
                        <span
                          className="font-12 totalRatio"
                        >
                          CHARTS.DONUT.TOOLTIP_HEAD
                        </span>
                        <span
                          className="uk-tooltip-cutom tooltip-text"
                        >
                          <i
                            className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                          />
                          <div
                            className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                          >
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              <strong
                                className="color-grey-blue-95"
                              >
                                CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                              </strong>
                              Â  
                              CHARTS.DONUT.TOOLTIP_SUB_TEXT
                            </span>
                            <span
                              className="color-red-60 mt-15 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION
                            </span>
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION_LHS
                               = Y (
                              LABELS.FORECAST
                              ) / X (
                              LABELS.PLAN
                              )
                            </span>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-24"
  >
    <h6
      className="m-0"
    >
      ANNUAL_TARGET.AVAILABLITY
    </h6>
    <div
      className="uk-card mt-24"
      uk-card="true"
    >
      <div
        className="uk-grid-small uk-grid ml-0"
        uk-grid="true"
      >
        <div
          className="uk-margin-bottom@s uk-width-3-4@m uk-padding-remove-left"
        >
          <div
            className="uk-flex-middle"
          />
        </div>
        <div
          className="uk-margin-bottom@s uk-width-1-4@m uk-padding-small mt-50"
        >
          <div
            className="uk-flex"
          >
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.PLAN
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.FORECAST
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
          </div>
          <div
            className="uk-flex uk-flex-center undefined"
          >
            <div
              className="mt-30"
            >
              <div
                className="uk-flex uk-flex-center"
              >
                <div
                  className="chart-continer"
                  style={
                    Object {
                      "height": 170,
                      "width": 170,
                    }
                  }
                >
                  <div
                    className="dount-chart-container"
                  >
                    <div />
                  </div>
                  <div
                    className=" donutmiddleData"
                  >
                    <div
                      className="uk-text-center thumb-font"
                    >
                      <div
                        className="jera-dislike-font"
                      >
                        <i
                          className="jera color-red-60 jera-dislike"
                        />
                      </div>
                      <div
                        className="font-prvalue color-red-60"
                      >
                        <span>
                          0
                        </span>
                        <span
                          className="font-12 vertical-top"
                        >
                          %
                        </span>
                      </div>
                      <div>
                        <span
                          className="font-12 totalRatio"
                        >
                          CHARTS.DONUT.TOOLTIP_HEAD
                        </span>
                        <span
                          className="uk-tooltip-cutom tooltip-text"
                        >
                          <i
                            className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                          />
                          <div
                            className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                          >
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              <strong
                                className="color-grey-blue-95"
                              >
                                CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                              </strong>
                              Â  
                              CHARTS.DONUT.TOOLTIP_SUB_TEXT
                            </span>
                            <span
                              className="color-red-60 mt-15 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION
                            </span>
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION_LHS
                               = Y (
                              LABELS.FORECAST
                              ) / X (
                              LABELS.PLAN
                              )
                            </span>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-24"
  >
    <h6
      className="m-0"
    >
      LABELS.THERMAL_EFFICIENCY
    </h6>
    <div
      className="uk-card mt-24"
      uk-card="true"
    >
      <div
        className="uk-grid-small uk-grid ml-0"
        uk-grid="true"
      >
        <div
          className="uk-margin-bottom@s uk-width-3-4@m uk-padding-remove-left"
        >
          <div
            className="uk-flex-middle"
          />
        </div>
        <div
          className="uk-margin-bottom@s uk-width-1-4@m uk-padding-small mt-50"
        >
          <div
            className="uk-flex"
          >
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.PLAN
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.FORECAST
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
          </div>
          <div
            className="uk-flex uk-flex-center undefined"
          >
            <div
              className="mt-30"
            >
              <div
                className="uk-flex uk-flex-center"
              >
                <div
                  className="chart-continer"
                  style={
                    Object {
                      "height": 170,
                      "width": 170,
                    }
                  }
                >
                  <div
                    className="dount-chart-container"
                  >
                    <div />
                  </div>
                  <div
                    className=" donutmiddleData"
                  >
                    <div
                      className="uk-text-center thumb-font"
                    >
                      <div
                        className="jera-dislike-font"
                      >
                        <i
                          className="jera color-red-60 jera-dislike"
                        />
                      </div>
                      <div
                        className="font-prvalue color-red-60"
                      >
                        <span>
                          0
                        </span>
                        <span
                          className="font-12 vertical-top"
                        >
                          %
                        </span>
                      </div>
                      <div>
                        <span
                          className="font-12 totalRatio"
                        >
                          CHARTS.DONUT.TOOLTIP_HEAD
                        </span>
                        <span
                          className="uk-tooltip-cutom tooltip-text"
                        >
                          <i
                            className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                          />
                          <div
                            className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                          >
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              <strong
                                className="color-grey-blue-95"
                              >
                                CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                              </strong>
                              Â  
                              CHARTS.DONUT.TOOLTIP_SUB_TEXT
                            </span>
                            <span
                              className="color-red-60 mt-15 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION
                            </span>
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION_LHS
                               = Y (
                              LABELS.FORECAST
                              ) / X (
                              LABELS.PLAN
                              )
                            </span>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    className="mt-24"
  >
    <h6
      className="m-0"
    >
      ANNUAL_TARGET.HEAT_RATE
    </h6>
    <div
      className="uk-card mt-24"
      uk-card="true"
    >
      <div
        className="uk-grid-small uk-grid ml-0"
        uk-grid="true"
      >
        <div
          className="uk-margin-bottom@s uk-width-3-4@m uk-padding-remove-left"
        >
          <div
            className="uk-flex-middle"
          />
        </div>
        <div
          className="uk-margin-bottom@s uk-width-1-4@m uk-padding-small mt-50"
        >
          <div
            className="uk-flex"
          >
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.PLAN
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
            <div
              className="uk-flex-left uk-width-1-2 uk-text-center"
            >
              <label
                className="font-14"
              >
                LABELS.FORECAST
              </label>
              <p
                className="mb-10 mt-5 font-14"
              >
                <span
                  className=""
                >
                  <strong>
                    
                    
                    
                  </strong>
                </span>
              </p>
            </div>
          </div>
          <div
            className="uk-flex uk-flex-center undefined"
          >
            <div
              className="mt-30"
            >
              <div
                className="uk-flex uk-flex-center"
              >
                <div
                  className="chart-continer"
                  style={
                    Object {
                      "height": 170,
                      "width": 170,
                    }
                  }
                >
                  <div
                    className="dount-chart-container"
                  >
                    <div />
                  </div>
                  <div
                    className=" donutmiddleData"
                  >
                    <div
                      className="uk-text-center thumb-font"
                    >
                      <div
                        className="jera-dislike-font"
                      >
                        <i
                          className="jera color-red-60 jera-dislike"
                        />
                      </div>
                      <div
                        className="font-prvalue color-red-60"
                      >
                        <span>
                          0
                        </span>
                        <span
                          className="font-12 vertical-top"
                        >
                          %
                        </span>
                      </div>
                      <div>
                        <span
                          className="font-12 totalRatio"
                        >
                          CHARTS.DONUT.TOOLTIP_HEAD
                        </span>
                        <span
                          className="uk-tooltip-cutom tooltip-text"
                        >
                          <i
                            className="jera-info-circle icn-16 color-blue-70 top-2 ml-3"
                          />
                          <div
                            className="uk-tooltip uk-animation-scale-up uk-animation-center uk-transdorm-origin-center-left uk-togglabe-enter uk-tooltip-bottom uk-tooltip-bottom-center uk-transform-origin-center-left uk-text-left font-10"
                          >
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              <strong
                                className="color-grey-blue-95"
                              >
                                CHARTS.DONUT.TOOLTIP_MAIN_TEXT
                              </strong>
                              Â  
                              CHARTS.DONUT.TOOLTIP_SUB_TEXT
                            </span>
                            <span
                              className="color-red-60 mt-15 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION
                            </span>
                            <span
                              className="color-grey-blue-80 uk-display-block"
                            >
                              CHARTS.DONUT.EQUATION_LHS
                               = Y (
                              LABELS.FORECAST
                              ) / X (
                              LABELS.PLAN
                              )
                            </span>
                          </div>
                        </span>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</li>
`;
